File: Python/number-of-boomerangs.py
class Solution:
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        num = 0

        for i, pointA in enumerate(points):
            d = dict()
            for j, pointB in enumerate(points):
                if i == j:
                    continue

                distance = (pointB[0] - pointA[0]) * (pointB[0] - pointA[0]) + (pointB[1] - pointA[1]) * (pointB[1] - pointA[1])

                if distance in d:
                    d[distance] = d[distance] + 1
                else:
                    d[distance] = 1

            for key in d.keys():
                num += d[key] * (d[key] - 1)

        return num

File: Swift/number-of-boomerangs.swift
class Solution {
    func numberOfBoomerangs(_ points: [[Int]]) -> Int {
        var num = 0

        for (i, pointA) in points.enumerated() {
            var dict = [Int: Int]()
            for (j, pointB) in points.enumerated() {
                if i == j {
                    continue
                }

                let distance = (pointB[0] - pointA[0]) * (pointB[0] - pointA[0])  + (pointB[1] - pointA[1]) * (pointB[1] - pointA[1])

                if let count = dict[distance] {
                    dict[distance] = count + 1
                } else {
                    dict[distance] = 1
                }
            }

            for key in dict.keys {
                num += dict[key]! * (dict[key]! - 1)
            }
        }

        return num
    }
}
